src/Build/Definition/Project.cs

```diff
@@ -3534,7 +3538,11 @@ private List<ProjectItem> AddItemHelper(ProjectItemElement itemElement, string u
             {
                 var itemFactory = new ProjectItemFactory(Owner, itemElement);
 
-                List<ProjectItem> items = Evaluator<ProjectProperty, ProjectItem, ProjectMetadata, ProjectItemDefinition>.CreateItemsFromInclude(DirectoryPath, itemElement, itemFactory, unevaluatedInclude, _data.Expander);
+                (List<ProjectItem> items, FileMatcher.SearchAction action, string fileSpecEscaped) = Evaluator<ProjectProperty, ProjectItem, ProjectMetadata, ProjectItemDefinition>.CreateItemsFromInclude(DirectoryPath, itemElement, itemFactory, unevaluatedInclude, _data.Expander);
+                if (action == FileMatcher.SearchAction.ReturnLogDriveEnumerationWildcard)
+                {
```

Comment:
TODO: Write test to reach this codepath

Reply:
See ProjectItem_Tests.cs. Test case passes for exception, but results in UnauthorizedAccessException upon logging a warning (happens after calling `GetFilesRecursive(...)` in FileMatcher.cs)